vumi/message.py

```diff
@@ -27,12 +27,17 @@ def format_vumi_date(timestamp):
 def parse_vumi_date(value):
     """Parse a timestamp string using the Vumi date format.
 
+    Timestamps without microseconds are also parsed correctly.
+
     :param str value:
         The string to parse.
     :return datetime:
         A datetime object representing the timestamp.
     """
-    return datetime.strptime(value, VUMI_DATE_FORMAT)
+    date_format = VUMI_DATE_FORMAT
+    if "." not in value:
+        date_format = date_format.split(".")[0]
```

Comment:
I'm not wild about this way of detecting milliseconds and splitting the format. Maybe a ``len(value) <= ...`` check would be better? And perhaps we can just have a (possibly `_` private) constant for `_VUMI_DATE_FORMAT_NO_MS`?